org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -1343,6 +1376,35 @@ protected void load (int ID, String trxName)
 		}
 	}	//	load
 
+	/**
+	 * Load record with UUID
+	 * 
+	 * @param uuID    UUID
+	 * @param trxName transaction name
+	 */
+	public void loadByUU(String uuID, String trxName)
+	{
+		// reset new values
+		m_newValues = new Object[get_ColumnCount()];
+		m_isUUID = true;
+		m_uuid = uuID;
+		checkImmutable();
+
+		if (log.isLoggable(Level.FINEST))
+			log.finest("uuID=" + uuID);
+		if (!Util.isEmpty(m_uuid, true))
+		{
+			setKeyInfo();
+			load(trxName);
+		}
+		else
+		{
+			loadDefaults();
+			m_createNew = true;
+			setKeyInfo();
+			loadComplete(true);
```

Comment:
should refactor and share the code instead of just duplicate this else part from the "load (int ID, String trxName)" method.